File: gcd.f90
PROGRAM GCD

  IMPLICIT NONE
  INTEGER :: A, B, C

  A = 54
  B = 24

  IF (A .LT. B) THEN
     C = A
     A = B
     B = C
  END IF

  DO
     C = MOD(A, B)
     IF (C .EQ. 0) EXIT
     A = B
     B = C
  END DO

  WRITE(*,*) 'GCD of 54 and 24 is ', B

END PROGRAM GCD

File: gcd.pas
program GCD (Input, Output);

var a, b, c, result, quit : Integer;

begin

   a := 54;
   b := 24;
   result := 0;
   quit := 0;

   if a < b then
   begin
      c := a;
      a := b;
      b := c;
   end;

   repeat
      c := a mod b;

      if c = 0 then
      begin
	 quit := 1;
	 result := b;
      end;

      a := b;
      b := c;
   until quit = 1;

   if result = 0 then result := b;

   writeln('GCD of 54 and 24 is ', result);
   
end.
